src/core/thread/address_resolver.cpp

```diff
@@ -365,10 +369,28 @@ Error AddressResolver::UpdateCacheEntry(const Ip6::Address &aEid, Mac::ShortAddr
     return error;
 }
 
-void AddressResolver::AddSnoopedCacheEntry(const Ip6::Address &aEid, Mac::ShortAddress aRloc16)
+void AddressResolver::UpdateSnoopedCacheEntry(const Ip6::Address &aEid,
+                                              Mac::ShortAddress   aRloc16,
+                                              Mac::ShortAddress   aDest)
 {
-    uint16_t    numNonEvictable = 0;
-    CacheEntry *entry;
+    uint16_t          numNonEvictable = 0;
+    CacheEntry *      entry;
+    Mac::ShortAddress macAddress;
+
+    VerifyOrExit(UpdateCacheEntry(aEid, aRloc16) != kErrorNone);
+
+    VerifyOrExit(Get<Mle::MleRouter>().IsFullThreadDevice());
```

Comment:
nit: reorder to exit earlier if not FTD.

```suggestion
    VerifyOrExit(Get<Mle::MleRouter>().IsFullThreadDevice());

    VerifyOrExit(UpdateCacheEntry(aEid, aRloc16) != kErrorNone);
```